dkg-gadget/src/worker.rs

```diff
@@ -1224,21 +1319,32 @@ where
 			DKGMsgPayload::Offline(..) | DKGMsgPayload::Vote(..) => {
 				let msg = Arc::new(dkg_msg);
 				let async_index = msg.msg.payload.get_async_index();
-				dkg_logging::debug!(target: "dkg_gadget::worker", "Received message for async index {}", async_index);
+				self.logger.debug(format!("Received message for async index {async_index}"));
 				if let Some(Some(rounds)) = self.signing_rounds.read().get(async_index as usize) {
-					dkg_logging::debug!(target: "dkg_gadget::worker", "Message is for signing execution in session {}", rounds.session_id);
+					self.logger.debug(format!(
+						"Message is for signing execution in session {}",
+						rounds.session_id
+					));
 					if rounds.session_id == msg.msg.session_id {
-						dkg_logging::debug!(target: "dkg_gadget::worker", "Message is for this signing execution in session: {}", rounds.session_id);
+						self.logger.debug(format!(
+							"Message is for this signing execution in session: {}",
+							rounds.session_id
+						));
 						if let Err(err) = rounds.deliver_message(msg) {
 							self.handle_dkg_error(DKGError::CriticalError {
 								reason: err.to_string(),
 							})
 						}
 					} else {
-						dkg_logging::debug!(target: "dkg_gadget::worker", "Message is for another signing round: {}", rounds.session_id);
+						self.logger.error(format!(
+							"Message is for another signing round: {}",
+							rounds.session_id
+						));
+						panic!("Message is for another signing round: {}", rounds.session_id)
```

Comment:
We should avoid panics please. When these cases happen we should identify the reasons in which they happen, can it be caused by malicious behaviour? Can we fail gracefully / ignore the message? If we panic the nodes stop and this becomes a DOS-able issue. If I can cause your node to crash then I will do so and you will be slashed.